client/src/ingredients/browser-compatibility-table/browser-support-notes.js

```diff
@@ -32,7 +45,7 @@ export function BrowserSupportNotes({ indexNote, blockElementTag, noteElementTag
   } else if (indexNote.notes.length > 0) {
     browserSupportNotes.push(
       indexNote.notes.map((note, index) => {
-        currentNoteContent = `<abbr class="only-icon" title="See implementation notes"><span>Notes</span><i class="ic-footnote"></i></abbr> ${note}`;
+        currentNoteContent = <><abbr className="only-icon" title="See implementation notes"><span>Notes</span><i className="ic-footnote"></i></abbr><span dangerouslySetInnerHTML={{__html: note}}></span></>;
```

Comment:
What are these `note` strings that have to contain HTML strings? It makes me nervous. `dangerouslySetInnerHTML` really is dangerous because if someone manages to trick the BCD data to be something very different, we have a security risk. 
The blobs of HTML we get from the prose in stumptown-content is also dangerous but it's something we're very aware of and there's plenty (at least will be) scripting that asserts that they're always safe.

Reply:
Example: Defaults to <code>metadata</code> in Chrome 64.

Reply:
I ended up adding in https://github.com/remarkablemark/html-react-parser to handle the markdown in the notes. Let me know if you're alright with that.

Reply:
No, that is not correct. The renderer shouldn't need to do that kind of work. That's what the `build-json` script in `stumptown-content` is supposed to own. 
If the BCD content in the JSON files that it picks up from `stumptown/packaged/**/*.json` isn't right, then let's hop over to https://github.com/mdn/stumptown-content/issues/new and resolve it there. 
File the issue and to avoid losing steam come back here and write in the issue URL as a code comment. 

Right, @wbamberg ?

Reply:
build-json just fetches the relevant BCD content from [browser-compat-data](https://github.com/mdn/browser-compat-data), given the BCD query string in the stumptown content. It does not do any processing on the BCD content.

The BCD content is JSON, but notes and descriptions are allowed to use a small subset of HTML tags (see e.g. https://github.com/mdn/browser-compat-data/blob/master/schemas/compat-data-schema.md#notes) and should be treated as HTML (as they already are in Kuma, via the Compat.ejs macro). BCD content should be at least as safe as stumptown content.

But I'm puzzled by "to handle the markdown in the notes". I don't think the notes should contain Markdown, can you show me where that happens?

Reply:
Sorry @wbamberg, i meant HTML markup, but the notes contain `<code>` tags.

Reply:
Yes, that's expected as per https://github.com/mdn/browser-compat-data/blob/master/schemas/compat-data-schema.md#notes.

Reply:
Being that it is expected that there could be a few HTML tags in the notes, is this okay?

Reply:
I think the expectation is that they are treated as HTML, as they are on MDN currently (see for example the `<code>` styling in the note about `auto` for IE here: https://developer.mozilla.org/en-US/docs/Web/CSS/margin#Browser_compatibility).

Reply:
@wbamberg right, if i just render the note without using the parser or `dangerouslySetInnerHTML` it looks like this:
```
<><abbr className="only-icon" title="See implementation notes"><span>Notes</span><i className="ic-footnote"></i></abbr>{note}</>;
```
![Screen Shot 2019-08-21 at 12 04 21 PM](https://user-images.githubusercontent.com/1201062/63448507-068bfd00-c40c-11e9-912c-18ead304f69b.png)

When i use the parser it looks like the expected result:
```
<><abbr className="only-icon" title="See implementation notes"><span>Notes</span><i className="ic-footnote"></i></abbr>{Parser(note)}</>;
```
![Screen Shot 2019-08-21 at 12 05 04 PM](https://user-images.githubusercontent.com/1201062/63448543-19063680-c40c-11e9-8195-8c3ea0e789a9.png)

Just to confirm, do you want me to use `dangerouslySetInnerHTML` or keep the `html-react-parser`? The `html-react-parser` will strip <script> tags even though it is doubtful that one would ever end up in the notes.

Reply:
Thanks for the clarification @joedarc !

> The blobs of HTML we get from the prose in stumptown-content is also dangerous but it's something we're very aware of and there's plenty (at least will be) scripting that asserts that they're always safe.

As this implies, we already use `dangerouslySetInnerHTML` for the other HTML content, and  I'd say that BCD content is at least as safe as stumptown-content.

But there is a general question about what to do about HTML we receive from relatively-trustworthy-sources like BCD and stumptown-content. Both these sources should have their own sanitizing but the renderer might want to sanitize as well, just in case.

I think it's @peterbe 's decision how to render content, but my 2c: we should use `dangerouslySetInnerHTML` here, but sanitize the content first, and we should do the same thing anywhere else we use `dangerouslySetInnerHTML`.

Reply:
Definitely `dangerouslySetInnerHTML` in favor of any runtime moving parts that could be avoided. For example, if we, for some reason, can't trust this HTML to be dangerously injected we can write a thing in the `stumptown-renderer/cli` that does the heavy lifting of making us more comfortable. 

My fear is that I'm not familiar with how the BCD pipeline and checks-and-balances work. If someone "sneaks" in a PR on BCD and changes the note there to `Defaults to <code>metadat</code> in Chrome <script>alert(xss)</script>` without it being caught by humans or automation there. 

For the sake of this PR, just use `dangerouslySetInnerHTML` and we'll file some issues about investigating or writing extra checks that makes sure BCD and stumptown-content are safe to dangerously inject.

Reply:
By the way, it is my intention to write a tool that ships stumptown-content pull requests into its own staging subdomains. First of all, those subdomains will never be hosted on `.mozilla.org` anyway. 

Second, if a PR is turned into a fully working subdomain (e.g. pr-1234.stumptown-staging.io) it means it would build without a human approver and that'd make a great test of trying to break the system. The second worst that can happen is that someone writes a rogue site that hacks into all `*.stumptown-staging.io` pages. The worst thing that can happen is that someone manages to put up a `.stumptown-staging.io` site installs malware into unsuspecting browsers. I.e. "Here go to this Mozilla site. It's fine. It's made by Muzilla."

Reply:
> For the sake of this PR, just use dangerouslySetInnerHTML

Reverted back to the original implementation per request. 👍

Reply:
> My fear is that I'm not familiar with how the BCD pipeline and checks-and-balances work. If someone "sneaks" in a PR on BCD and changes the note there to `Defaults to <code>metadat</code> in Chrome <script>alert(xss)</script>` without it being caught by humans or automation there.

I'm happy to prioritize fixing this on the BCD side.